src/Mod/Path/PathScripts/PathToolBitGui.py

```diff
@@ -181,16 +181,31 @@ def Create(self, name='ToolBit', shapeFile=None):
         FreeCAD.ActiveDocument.commitTransaction()
         return tool
 
+def isValidFile(filename):
```

Comment:
This has the side effect of actually creating that (empty) file if it does not exist. That might be undesirable in some cases. I'm also not sure about the function name: "valid" can mean a lot of things when you are taking about a file. Maybe `isWritableFile`?

Reply:
The intent isn't just to check if it's writable.  I found from my own use the tempation to create toolbits that have names corresponding to size.  In US, that's commonly a fraction  "3/4 inch 4 flute endmill."   Fraction, obviously are a problem and threw an ugly error.  So I've renamed the function to isValidFileName.   
You're right that it creates the empty file but I don't think that's an issue in this case.  Because it's used specifically in the case of creating the file.  If the filename is valid and can be written at all, the empty file is immediately overwritten with the toolbit attributes.

Reply:
one could argue that you mihgt wanna buy real tools - instead of this "fraction" business ;)